contextvh/src/main/java/contextvh/ContextEnv.java

```diff
@@ -107,7 +110,8 @@ protected Percept performEntityAction(final String entity,
      */
     @Override
     public TygronEntity createNewEntity(final EntityListener listener, final String stakeholder, final Integer slot) {
-        return new ContextEntity(listener, stakeholder, slot);
+    	entity =  new ContextEntity(listener, stakeholder, slot);
```

Comment:
In the case of our agents this is fine but what happens if we make a mas file with two stakeholders in it?